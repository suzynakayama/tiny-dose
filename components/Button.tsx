import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import React from 'react';

// This component is a reusable button that can be used throughout the app
// It accepts an onPress function, text to display, and optional styles for the button and text
// It uses TouchableOpacity for a better user experience with touch feedback

interface ButtonProps {
  onPress: () => void;
  text: string;
  buttonStyle?: object;
  textStyle?: object;
}

export default function Button({
  onPress,
  text,
  buttonStyle,
  textStyle,
}: ButtonProps) {
  return (
    <TouchableOpacity
      accessibilityRole='button'
      accessibilityLabel={text}
      accessibilityHint='Press to perform an action'
      style={[styles.button, buttonStyle]}
      onPress={onPress}
    >
      <Text style={[styles.buttonText, textStyle]}>{text}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#01CDD4',
    paddingVertical: 12,
    paddingHorizontal: 5,
    borderRadius: 30,
    alignItems: 'center',
    shadowColor: '#031F72',
    shadowOffset: { width: 4, height: 6 },
    shadowOpacity: 0.25,
    elevation: 6,
  },
  buttonText: {
    color: '#fff',
    fontSize: 18,
    fontWeight: 'bold',
    textShadowColor: '#031F72',
    textShadowOffset: { width: 1, height: 1 },
    textShadowRadius: 2,
  },
});
