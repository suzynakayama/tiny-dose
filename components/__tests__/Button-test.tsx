import React from 'react'
import { render, screen } from '@testing-library/react-native'
import Button from '../Button'

test('renders correctly', () => {
  const ButtonComponent = render(<Button text='Click Me' onPress={() => {}} />)
  const button = screen.getByRole('button')
  expect(button).toBeOnTheScreen()
  expect(button).toHaveProp(
    'style',
    expect.objectContaining({
      alignItems: 'center',
      backgroundColor: '#01CDD4',
      borderRadius: 30,
      opacity: 1,
      paddingHorizontal: 5,
      paddingVertical: 12,
      shadowColor: '#031F72',
      shadowOffset: { height: 6, width: 4 },
      shadowOpacity: 0.25,
    })
  )

  // Snapshot testing
  const buttonJson = ButtonComponent.toJSON()
  expect(buttonJson).toMatchSnapshot()
})

test('received styles correctly', () => {
  render(<Button text='Click Me' onPress={() => {}} buttonStyle={{ backgroundColor: 'red' }} />)
  const button = screen.getByRole('button')
  expect(button.props.style).toHaveProperty('backgroundColor', 'red')
})
