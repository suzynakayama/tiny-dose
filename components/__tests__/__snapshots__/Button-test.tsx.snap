// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityHint="Press to perform an action"
  accessibilityLabel="Click Me"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#01CDD4",
      "borderRadius": 30,
      "elevation": 6,
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 12,
      "shadowColor": "#031F72",
      "shadowOffset": {
        "height": 6,
        "width": 4,
      },
      "shadowOpacity": 0.25,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textShadowColor": "#031F72",
          "textShadowOffset": {
            "height": 1,
            "width": 1,
          },
          "textShadowRadius": 2,
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;
